openspoor/visualisations/trackmap.py

```diff
@@ -17,60 +19,68 @@ class PlotObject(ABC):
     A parent class for every object that can be plotted on a TrackMap
     """
 
+    def __init__(self, data, popup):
+        if popup is None or isinstance(popup, list):
+            self.popup = popup
+        elif isinstance(popup, str):
+            self.popup = [popup]
+        else:
+            TypeError('Unknown type for popup in linestrings')
+
+        if isinstance(data, gpd.GeoDataFrame):
+            self.data = data
+        elif isinstance(data, str):
+            self.data = gpd.read_file(data)
+        else:
+            raise TypeError('Provide either a geopandas dataframe or a file location of a csv to show')
+        self.data = self.data.to_crs('EPSG:4326')
+        if self.popup:
```

Comment:
This if statement is unnecessary because of the if else above

Reply:
self.popup might still be None; I did change it to explicitly check for this, it is not clear from reading it here.

Reply:
I see there is no 'raise' in the TypeError above in line 28, maybe thats why it still can be none?

Reply:
As discussed, self.popup can be set to the value None, which is why it can reach up to this point. This is now made more explicit in the check on this line.